### PR TITLE
RavenDB-12355 limit database paths

### PR DESCRIPTION
**Pull request by iftahbe**

### CODE REVIEW

**Review by ppekrol — APPROVED** [review]

File: src/Voron/Util/Settings/PathSettingBase.cs [hunk 1]
```diff
@@ -94,5 +94,25 @@ public static string ToFullPath(string inputPath, string baseDataDirFullPath)
...
+        {
+            var rootDirInfo = new DirectoryInfo(rootPath);
+            var userDirInfo = new DirectoryInfo(userPath);
+
+            if (userDirInfo.FullName == rootDirInfo.FullName)
+                return true;
+
+            while (userDirInfo.Parent != null)
+            {
+                if (userDirInfo.Parent.FullName == rootDirInfo.FullName)
```
Comment (ayende) [inline src/Voron/Util/Settings/PathSettingBase.cs]: I wonder if we need to make this a case insensitive comparison (on Windows, at least?)

File: src/Raven.Server/Web/System/AdminDatabasesHandler.cs [hunk 2]
```diff
@@ -227,6 +230,17 @@ public async Task Put()
                     throw new ArgumentException("DatabaseName property has invalid value (null, empty or whitespace only)");
                 databaseRecord.DatabaseName = databaseRecord.DatabaseName.Trim();
 
+                if (ServerStore.Configuration.Core.EnforceDataDirectoryPath 
+                    && databaseRecord.Settings.TryGetValue("DataDir", out var dir))
```
Comment (ppekrol) [inline src/Raven.Server/Web/System/AdminDatabasesHandler.cs]: Use `RavenConfiguration.GetKey(x => x.Core.DataDir)`

**Review by ayende — APPROVED** [review]

**Comment by ayende** [comment]
The test failure is a check to show that a all admin routes have `admin` in them. Need to handle that somehow.
I don't think we have any backward compatibility concerns here, though.

**Comment by ppekrol** [comment]
This is only used by studio so it can be changed to `/admin/studio-tasks/full-data-directory` without a problem.

**Pull request merged**